### PR TITLE
Change eam input, section name, and observational/reference data paths

### PR DESCRIPTION
**Pull request by hsiangheleellnl**
There are some changes regarding eam file reading and the section name change from e3sm_chem_diags to chemdyg.

### CODE REVIEW

**Comment by xylar** [comment]
@hsiangheleellnl, it seems like you'll need to rebase this branch onto `origin/main` to pick up the files getting moved in #1.  Other than that, I'm not sure I can provide much useful feedback on the code changes themselves.

**Comment by hsiangheleellnl** [comment]
rebase the changes onto 'origin/main'

**Comment by xylar** [comment]
@hsiangheleellnl, there's no need to close the PR.  You can just rebase onto `origin/master` and then force-push the branch to the remote branch.  Then, the PR will be updated automatically.

**Comment by xylar** [comment]
@hsiangheleellnl, let me know if more detailed instructions would be helpful.  I'm happy to provide them.

**Comment by tangq** [comment]
@xylar , I showed Hsiang-He how to rebase and the conflicts should be resolved now.

The commit ([832b549](https://github.com/E3SM-Project/ChemDyg/pull/6/commits/832b5493255c556941f31dbbbc11431e3c68950e)) changes to use relative paths for input files as suggested in https://github.com/E3SM-Project/ChemDyg/issues/4#issuecomment-1470790434.

The `reference_data_path` is set in `ChemDyg_example_script.cfg` and should match the root input data path, which `mache` should know for each machine. Do you have suggestions on how to set the root path automatically for different machines?

**Comment by xylar** [comment]
@tangq, thank you for clarifying what you need from me here.  I can add an example of how to use `mache` to find the path on each machine in one of your bash scripts and you can hopefully figure out how to make similar changes to other scripts.

File: ChemDyg_example_script.cfg [hunk 1]
```diff
@@ -55,10 +55,11 @@ active = False
...
 active = False
 
-[e3sm_chem_diags]
+[chemdyg]
 active = True
-short_name = '20230221.F20TR.NGD_v3atm_v2.1oi_alt'
-years = "1985:1999:15",
+short_name = '20230307.amip.NGD_v3atm.chem-mam5-vbs.chrysalis'
+years = "2013:2013:1",
+reference_data_path = /lcrc/group/acme/public_html/diagnostics/observations/Atm/ChemDyg_inputs
```
Comment (xylar) [inline ChemDyg_example_script.cfg]: I would not provide the `reference_data_path` as a config option at all.  I would use `mache` to figure that out, as I will demonstrate as soon as I am able.

**Comment by xylar** [comment]
@hsiangheleellnl and @tangq, I pushed a commit that I think will take care of this.  I think that zppy will provide the `diagnostics_base_path` config option (which it will get from `mache`) if I understand https://github.com/E3SM-Project/zppy/pull/402 correctly.

**Comment by tangq** [comment]
@xylar , thank you. The `diagnostics_base_path` config option is exactly what I am looking for.

**Review by tangq — APPROVED** [review]

**Pull request merged**